src/prefect/client/client.py

```diff
@@ -573,8 +573,8 @@ def register(
                 "Flows with required parameters can not be scheduled automatically."
             )
         if any(e.key for e in flow.edges) and flow.result_handler is None:
-            raise ClientError(
-                "Flows are required to have a result handler for storing inputs and outputs."
+            warnings.warn(
+                "It is strongly recommended you include a result handler on your Flow for certain Cloud features to work properly."
```

Comment:
Note for the reviewer: I'd like to improve this message / make it more informative and / or stronger

Reply:
Could you add a small explanation to the docs about this? Like the deployment considerations (https://docs.prefect.io/cloud/recipes/deployment.html) for example. That way this message could also have a link to more information